### PR TITLE
Fix non-utf8 reply parsing causing segmentation fault in Python 3

### PR DESCRIPTION
**Pull request by un-def**
It's possible to create Redis error replies that cannot be decoded with 'utf-8' codec in Python3 and cause `UnicodeDecodeError`.

For example:

```
EVAL 'æ' 0
-ERR Error compiling script (new function): user_script:1: unexpected symbol near '�'
```

Last three bytes (excluding whitespaces) are `27 c3 27`. 

```
>>> b'\x27\xc3\x27'.decode('utf-8')

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 1: invalid continuation byte
```

This case is not handled properly and causes segmentation fault. I've added some tests to demonstrate failure: [commit with new tests](https://github.com/un-def/hiredis-py/commit/c69d822caf2392c82fb42dd45349e2f22e12d6cc), [failed travis job](https://travis-ci.org/un-def/hiredis-py/builds/406579334).

This PR adds `UnicodeDecodeError` exception handling in Python 3 (raw undecoded `bytes` string will be used in case of utf-8 decode error).

### CODE REVIEW

File: src/reader.c [hunk 1]
```diff
@@ -107,7 +107,19 @@ static PyObject *createDecodedString(hiredis_ReaderObject *self, const char *str
...
+    /* "s" stands for "string" (string object, byte string) in Python 2
+     * and "str" (str object, unicode string) in Python 3 */
     PyObject *args = Py_BuildValue("(s#)", errstr, len);
+    if (args == NULL) {
+        #if IS_PY3K
+        /* Clear an error indicator and construct bytes object ("y") instead of str
+         * in case of UnicodeDecodeError (e.g., malformed utf-8 string) in Python 3 */
+        if (PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)) {
+            PyErr_Clear();
+            args = Py_BuildValue("(y#)", errstr, len);
```
Comment (ifduyue) [inline src/reader.c]: Is this a good idea? Behaviors are inconsistent. But I can't find a better solution.
What do you think about raising UnicodeDecodeError directly?

Comment (un-def) [reply src/reader.c]: I agree that this is an inconsistent behavior and this requires additional type check by the caller, but I think that `UnicodeDecodeError` is an even more unexpected behavior. Maybe decoding bytes to a string object in `replace` or `backslashreplace` mode would be a better solution?

Comment (ifduyue) [reply src/reader.c]: Decoding to string with `replace` mode on would be better, will work just like redis-cli.

Comment (un-def) [reply src/reader.c]: I've just opened new PR with suggested solution https://github.com/redis/hiredis-py/pull/73.

**Comment by un-def** [comment]
Closed in favor of https://github.com/redis/hiredis-py/pull/73.

**Pull request closed without merging**